lib/miq_automation_engine/service_models/miq_ae_service_manageiq-providers-redhat-infra_manager.rb

```diff
@@ -3,5 +3,6 @@ class MiqAeServiceManageIQ_Providers_Redhat_InfraManager < MiqAeServiceEmsInfra
     expose :supports_vm_import?
     expose :submit_import_vm
     expose :submit_configure_imported_vm_networks
+    expose :switches, :association => true
```

Comment:
In ActiveRecord the `switches` association is defined on the [ManageIQ::Providers::InfraManager](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/infra_manager.rb#L20) model which both of these classes derive from.  (Note: EmsInfra is an alias for ManageIQ::Providers::InfraManager here https://github.com/ManageIQ/manageiq/blob/master/app/models/aliases/ems_infra.rb)

Exposing `switches` should be done once on the intermediate service_model class `ManageIQ::Providers::InfraManager`.  It looks like this file needs to be created.

Also this `miq_ae_service_manageiq-providers-redhat-infra_manager.rb` should be changed to derive from `MiqAeServiceManageIQ_Providers_InfraManager` instead of `MiqAeServiceEmsInfra`.